src/index.js

```diff
@@ -6,14 +6,15 @@ const withIs = require('class-is')
 /**
  * Represents an immutable block of data that is uniquely referenced with a cid.
  *
- * @constructor
- * @param {Uint8Array} data - The data to be stored in the block as a Uint8Array.
- * @param {CID} cid - The cid of the data
- *
  * @example
  * const block = new Block(Uint8Array.from([0, 1, 2, 3]), new CID('...'))
  */
-class Block {
+module.exports = class Block {
```

Comment:
@rvagg you know way more about module exports than I do. I honestly don't know what all the implications are that this change has. Do you?

Reply:
Technically this is fine, it's just very weird! A double-assignment to a passed in variable. I guess the comment on the 2nd last line explains it, so 🤷, whatever. I'm not a huge fan of `withIs` anyway so this just adds to my frustration with that pattern and resolve to do better in next-gen APIs.

Reply:
The first `module.exports` here is to set the export type of the module.  (**The typings engine will only use the first type assignation to `module.exports`, i.e. tricking the typings engine**)
The last line is to wrap the class constructor with `withIs` but does not change the export type. (The return type for `withIs` is weird in TypeScript)

Reply:
Other approach:
```js
module.exports = Block
module.exports = withIs(Block, { className: 'Block', symbolName: '@ipld/js-ipld-block/block' })
```

In contrast to
```js
/** @type {typeof Block} */
module.exports = withIs(Block, { className: 'Block', symbolName: '@ipld/js-ipld-block/block' })
// cannot export the instance type and the class constructor type simultaneously
```

Reply:
yeah, it's fine as, I wouldn't mind expanding on that comment though, "to trick the typings engine" doesn't really say it clearly, just make it clear so that the next person in here refactoring doesn't just remove the weird duplication out of reflex.

Reply:
Maybe we should just cut `withIs`  dependency here as we did in `CID`s ? It would remove need for this workaround.